lib/action_client/callbacks/after_submit.rb

```diff
@@ -0,0 +1,60 @@
+module ActionClient
+  module Callbacks
+    class AfterSubmit
+      def initialize(with_status: nil, &block)
+        @with_status = with_status || (100..599)
+        @block = block
+      end
+
+      def call(status, headers, body)
+        http_statuses = Array(with_status).map do |status_code|
+          Rack::Utils.status_code(status_code)
+        end
+
+        if http_statuses.any? { |code| Array(code).include?(status) }
+          execute(status, headers, body)
+        else
+          [status, headers, body]
+        end
+      end
+
+      private
+
+      attr_reader :block, :with_status
+
+      def execute(status, headers, body)
+        if block.arity == 1
```

Comment:
I wonder if we could _replace conditional with polymorphism_, and move this arity check to decide which class we instantiate?

Reply:
The single-argument variant could maybe even be defined in terms of the multi-argument variant

Reply:
@georgebrock I've extracted some block classes (including HttpStatusFilter) in https://github.com/thoughtbot/action_client/pull/3/commits/c431bee0d982bccebeec7b9afd54d53f501afe5d.

I'd appreciate some feedback on how well I was able to execute _replace conditional with polymorphism_.